src/deroff.rs

```diff
@@ -538,12 +538,311 @@ impl Deroffer {
+                // return self.text();
+            }
+            "." => {
+                self.r#macro = false;
+                // self.condputs("\n");
+                return true;
+            }
+            _ => (),
+        };
+
+        self.nobody = false;
+        let s0s1 = self.s[..2].to_owned();
+
+        if self.g_macro_dispatch(s0s1.as_str()) {
+            return true;
+        }
+
+        if SKIP_HEADERS && self.nobody {
+            return true;
+        }
+
+        self.s = self.skip_leading_whitespace(&self.s).to_owned();
+        while !self.s.is_empty() && !Self::is_white(&self.s, 0) {
+            self.s = self.skip_char(&self.s, None).to_owned();
+        }
+
+        self.s = self.skip_leading_whitespace(&self.s).to_owned();
+
+        loop {
+            if !self.quoted_arg() && !self.text_arg() {
+                if !self.s.is_empty() {
+                    // self.condputs(self.str_at(0));
+                    self.condputs(Self::str_at(&self.s, 0));
+                    self.s = self.skip_char(&self.s, None).to_owned();
+                } else {
+                    return true;
+                }
+            }
+        }
+    }
+
+    fn font(&mut self) -> bool {
+        if let Some(m) = self.g_re_font.find(&self.s) {
+            self.s = self.skip_char(&self.s, Some(m.end())).to_owned();
+            true
+        } else {
+            false
+        }
+    }
+
+    fn comment(&self) -> bool {
+        unimplemented!()
+    }
+
+    fn numreq(&mut self) -> bool {
+        // In the python, it has a check that is already handled in esc_char_backslash, which is
+        // the only place it gets called, so I'll omit that check here
+
+        // This is written as `self.macro += 1` in the source, but I dont know why
+        // it does the same thing (false -> true, true -> still true) :shrug:
+        // self.r#macro = true;
+        // Upon further investigation, this is the weirdest function ever
+        // This is just a state placeholder thing
+        let mut m = self.r#macro as u8 + 1;
+
+        // self.skip_char(3);
+        self.s = self.skip_char(&self.s, Some(3)).to_owned();
+
+        // There's this, which has a comment explaining my thoughts right now
+        /*
+        while self.str_at(0) != "'" and self.esc_char():
+            pass  # Weird
+        */
+        // And I dont know what the purpose of it would even be, if you can tell, lmk
+
+        // if self.str_at(0) == '\'' {
+        if Self::str_at(&self.s, 0) == "'" {
+            // self.skip_char(1);
+            self.s = self.skip_char(&self.s, None).to_owned();
+        }
+
+        m -= 1;
+        self.r#macro = m != 0;
+
+        true
+    }
+
+    fn text_arg(&self) -> bool {
+        unimplemented!()
+    }
+
+    fn prch(&self, idx: usize) -> bool {
+        self.s
+            .get(idx..=idx)
+            .map(|c| !" \t\n".contains(c))
+            .unwrap_or_default()
```

Comment:
Should we parse just a single byte here instead of a string?

Reply:
we could go for a `char` instead of `str` but `u8`s are inadvisable.
revised code: 
```rust
self.s
    .chars()
    .nth(idx)
    .map(|c| !c.is_whitespace())
    .unwrap_or_default()
```

Note on the u8s thing: `char` is a type that wraps multi byte unicode code points, so if we went off of bytes, we could miss some non ascii (mutli byte) whitespace.